optimum/gptq/quantizer.py

```diff
@@ -41,13 +43,28 @@
 
 if is_auto_gptq_available():
     from auto_gptq import exllama_set_max_input_length
-    from auto_gptq.modeling._utils import autogptq_post_init
+    from auto_gptq.modeling._utils import autogptq_post_init as gptq_post_init
     from auto_gptq.quantization import GPTQ
-    from auto_gptq.utils.import_utils import dynamically_import_QuantLinear
+    from auto_gptq.utils.import_utils import dynamically_import_QuantLinear as hf_select_quant_linear
+
+
+if is_gptqmodel_available():
+    from gptqmodel import exllama_set_max_input_length
+    from gptqmodel.quantization import GPTQ
+    from gptqmodel.utils.importer import hf_select_quant_linear
+    from gptqmodel.utils.model import gptqmodel_post_init as gptq_post_init
 
 logger = getLogger(__name__)
 
 
+if not is_gptqmodel_available():
+    logger.warning("auto_gptq will be deprecated in the future, please `pip install gptqmodel` for gptq model")
```

Comment:
Don't put that here as it will probably trigger a warning for all users to tries to import this file that haven't installed `gptqmodel`